packages/vite/src/node/packages.ts

```diff
@@ -143,6 +143,20 @@ export function findNearestPackageData(
   return null
 }
 
+// Finds the nearest package.json with a `name` field
+export function findNearestMainPackageData(
+  basedir: string,
+  packageCache?: PackageCache,
+): PackageData | null {
+  const nearestPackage = findNearestPackageData(basedir, packageCache)
+  return (
+    nearestPackage &&
+    (nearestPackage.data.name
+      ? nearestPackage
+      : findNearestMainPackageData(path.dirname(basedir), packageCache))
```

Comment:
```suggestion
      : findNearestMainPackageData(path.dirname(nearestPackage.dir), packageCache))
```

Should the basedir be from the nearest package dir?

Reply:
Oh yes! Good catch! I thought about it before implementing it because this was one of the perf justification then I completely forgot 😅